lib/rules/constructor-super.js

```diff
@@ -274,15 +300,15 @@ module.exports = function(context) {
                         message: "Unexpected duplicate 'super()'.",
                         node: node
                     });
+                } else if (!funcInfo.superIsConstructor) {
+                    context.report({
+                        message: "Unexpected 'super()' because 'super' is not a constructor.",
```

Comment:
Let's go with this for now. If we get feedback that the message is confusing, we can always change it later.